docs/dev/gen-nb-src.py

```diff
@@ -0,0 +1,59 @@
+'''
+Transforms a .html notebook into its source .py/.scala/.r/,sql file.
+
+This script is used by the CircleCI job 'check-docs'. Before running this, configure
+your Databricks CLI profile.
+
+Example usage:
+  python3 docs/dev/gen-nb-src.py \
+    --html docs/source/_static/notebooks/etl/variant-data.html
+'''
+import click
+import subprocess
+import os
+import uuid
+
+NOTEBOOK_DIR = 'docs/source/_static/notebooks'
+SOURCE_DIR = 'docs/source/_static/zzz_GENERATED_NOTEBOOK_SOURCE'
+SOURCE_EXTS = ['scala', 'py', 'r', 'sql']
+
+
+@click.command()
+@click.option('--html', required=True, help='Path of the HTML notebook.')
+@click.option('--cli-profile', default='docs-ci', help='Databricks CLI profile name.')
```

Comment:
Might it make more sense to rely on the databricks CLI's default profile?

Reply:
I found this useful during manual testing, as I already have a default Databricks CLI profile and wanted to test this on the same shard as the CI.

Reply:
I mean that it makes sense to have this argument, but the default value should be the default profile. If you're running locally, I don't think there's any reason to expect a profile called `docs-ci` to exist. We can pass the `docs-ci` argument from our CI script.